turbopack/crates/turbo-tasks-macros/src/value_trait_macro.rs

```diff
@@ -119,18 +133,22 @@ pub fn value_trait(args: TokenStream, input: TokenStream) -> TokenStream {
                 }
                 // Add a dummy implementation that derefences the box and delegates to the
                 // actual implementation.
-                dynamic_trait_fns.push(if sig.asyncness.is_some() {
-                    quote! {
-                        #sig {
-                            let reference: &dyn #trait_ident = &*self;
-                            reference.#ident(#(#args),*).await
+                dynamic_trait_fns.push(match (is_vc_receiver, sig.asyncness.is_some()) {
+                    (true, true) | (true, false) => quote! { #item },
```

Comment:
```suggestion
                    (true, _) => quote! { #item },
```

But I don't really get what the macros change is doing...?

Reply:
Does it remove all turbo-tasks functions from all trait method with `self: Vc<Self>`?

That doesn't sounds right...?

Reply:
Could you at least split this change from the other changes into separate PRs?

Reply:
that isn't what it does.  

Currently, trait items without `turbo_task::function` are not turbotasks,  so the choice is clear (add the annotation, or not).  That has been the case for a few months now.

All this does is provide for the case on a non-turbotask function where the receiver type is `self: Vc<Self>` by doing less rewriting.

Reply:
ill split the macro change